core/src/main/java/com/linecorp/armeria/common/logging/LoggingDecoratorBuilder.java

```diff
@@ -258,21 +259,18 @@ public T responseCauseSanitizer(Function<? super Throwable, ?> responseCauseSani
     }
 
     /**
-     * Sets the rate at which to sample requests to log. Any number between {@code 0.0} and {@code 1.0} will
-     * cause a random sample of the requests to be logged. The random sampling is appropriate for low-traffic
-     * (ex servers that each receive &lt;100K requests). If unset, all requests will be logged.
+     * Sets the {@link Sampler} that to sample requests to log.
      */
-    public T samplingRate(float samplingRate) {
-        checkArgument(0.0 <= samplingRate && samplingRate <= 1.0, "samplingRate must be between 0.0 and 1.0");
-        this.samplingRate = samplingRate;
+    public T sampler(Sampler<? super RequestLog> sampler) {
+        this.sampler = sampler;
         return self();
     }
 
     /**
-     * Returns the rate at which to sample requests to log.
+     * Returns the {@link Sampler} that to sample requests to log.
```

Comment:
.. that determines which request needs logging.

Reply:
Thank you for review. Fixed in 0c5aae3.